### DISCUSSION TITLE
How should I set up vscode for contributions?

### DISCUSSION THREAD

**Discussion by ChloeSartorelli** [ddev/ddev#5778] [Q&A] [answered]
@rfay as per https://github.com/ddev/ddev/pull/5732#issuecomment-1925360779

> please see if you can convince your IDE to use standard gofmt formatting, which is what's generally required. As you point out, it makes a mess and is hard to review when things get changed that aren't relevant to the PR. Happy to chat with you about this, I know the timezones are hard. But it would be cool if you could experiment a bit to see if you can calm it down a bit. What IDE are you using? How do you have it set up? We can discuss in another issue if that's the easiest.

I'm using vscode with [this extension](https://marketplace.visualstudio.com/items?itemName=golang.Go) - I haven't don't any other configuration related to go.

**Comment by rfay** [comment]
I guess that's all we do in gitpod setup, https://github.com/ddev/ddev/blob/dab6258cf057132204f5ed8fd461c979f8ad1125/.gitpod.yml#L30-L40

I wonder if there's any gofmt-specific stuff to do. Do you have gofmt installed? Homebrew installs it as part of `go`. It's the definitive style formatter for go in general, and it's what DDEV insists on in the code tests.

**Comment by ChloeSartorelli** [comment]
Yup `gofmt` is installed.

> it's what DDEV insists on in the code tests

Code tests or lint tests? The lint test hasn't failed on me since my early contributions so I think maybe that's working as expected?

**Reply by rfay** [reply]
It's the linting test, and no it won't fail, but I'm hoping that your vscode uses gofmt as "source of truth".

**Reply by tyler36** [reply]
> Do you have gofmt installed?

For others that might be reading this. How can you check that gofmt is install?

**Reply by rfay** [reply]
`which gofmt` will show it if it's installed and in your path.

**Comment by rfay** [comment, marked as answer]
The changes that actually ended up in your PRs weren't a problem, and I'm not sure there's anything we need to do about them. In the past, we've had aggressive things from vscode (rearranging imports without being asked, etc.) but yours don't show that. I think that's done by additional vscode golang extensions. 

If you see anything that can be improved in https://github.com/ddev/ddev/tree/master/.vscode it's welcome. 

I don't think we have a problem, and thanks for your amazing contributions.

If you see suspect unneeded reformatting in your PRs (beyond what gofmt would do), please poke around to chase it, but again I think it's all fine.

**Reply by ChloeSartorelli** [reply]
Sweet, no worries. This is the only project I use go for and I'm still very new to it - it's unlikely I'd notice that formatting doesn't match what `gofmt` would produce, so if you notice any non-`gofmt`-compliant formatting creep in please let me know (pointing out the specific formatting in question) and I'll look into it.

**Comment by tyler36** [comment]
@GuySartorelli  

If you haven't already seen it, there was a contributor-training workshop: [Setting up a Go Development Environment](https://www.dropbox.com/scl/fi/gka3bwm3pwpchryg50l0t/20230725_contributor_training_go_environment.mp4?rlkey=3hlhugszdxi4hm6rmwlw28r5f&dl=0)

It cover some VSCode setup.

Other workshops are list here: https://ddev.com/blog/contributor-training/

**Reply by ChloeSartorelli** [reply]
Thanks, I'll give it a look and see if there's something I might want to change based on that.
